src/aspire/estimation/covar2d.py

```diff
@@ -101,29 +101,31 @@ def get_mean(self, coeffs, ctf_fb=None, ctf_idx=None):
         :param coeffs: A coefficient vector (or an array of coefficient vectors) to be averaged.
         :param ctf_fb: The CFT functions in the FB expansion.
         :param ctf_idx: An array of the CFT function indices for all 2D images.
+            If ctf_fb or ctf_idx is None, the identity filter will be applied.
         :return: The mean value vector for all images.
         """
         if coeffs.size == 0:
             raise RuntimeError('The coefficients need to be calculated!')
 
         if (ctf_fb is None) or (ctf_idx is None):
-            return self._get_mean(coeffs)
+            ctf_idx = np.zeros(coeffs.shape[1], dtype=int)
+            ctf_fb = [blk_diag_eye(blk_diag_partition(RadialCTFFilter().fb_mat(self.basis)))]
 
-        b = np.zeros((self.basis.count, 1), dtype=self.as_type)
+        b = np.zeros((self.basis.count, 1), dtype=coeffs.dtype)
 
-        A = blk_diag_zeros(blk_diag_partition(ctf_fb[0]), dtype=self.as_type)
+        A = blk_diag_zeros(blk_diag_partition(ctf_fb[0]), dtype=coeffs.dtype)
         for k in np.unique(ctf_idx[:]).T:
             coeff_k = coeffs[:, ctf_idx == k]
             weight = np.size(coeff_k, 1)/np.size(coeffs, 1)
-            mean_coeff_k = self._get_mean(coeff_k)
+            mean_coeff_k = self._get_mean(coeff_k).reshape((self.basis.count, 1))
             ctf_fb_k = ctf_fb[k]
             ctf_fb_k_t = blk_diag_transpose(ctf_fb_k)
             b = b + weight*blk_diag_apply(ctf_fb_k_t, mean_coeff_k)
```

Comment:
Ideally, we should have `blk_diag_apply` still work for vectors. For now, we can make an issue and replace `mean_coeff_k` by `mean_coeff_k[:, np.newaxis]`.

Reply:
Done. Open the issue #111 for reimplement blk_diag_apply and other functions.

Reply:
This still has `mean_coeff_k` as a 2D array. Replace it with a vector and change this line to
```
b = b + weight*blk_diag_apply(ctf_fb_k_t, mean_coeff_k[:, np.newaxis])
```

Reply:
The output of blk_diag_apply is a 2D array. If I do it for both, the format looks quite ugly. I think it is fine to keep it now since it is internal one. But if you really want it get changed I will do it.

Reply:
Done.